neo/rawio/intanbinaryrawio.py

```diff
@@ -0,0 +1,442 @@
+            )
+
+        # No events
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes
+        spike_channels = []
+        spike_channels = np.array(spike_channels, dtype=_spike_channel_dtype)
+
+        # fill into header dict
+        self.header = {}
+        self.header["nb_block"] = 1
+        self.header["nb_segment"] = [1]
+        self.header["signal_streams"] = signal_streams
+        self.header["signal_channels"] = signal_channels
+        self.header["spike_channels"] = spike_channels
+        self.header["event_channels"] = event_channels
+
+        self._generate_minimal_annotations()
+
+    def _segment_t_start(self, block_index, seg_index):
+        return 0.0
+
+    def _segment_t_stop(self, block_index, seg_index):
+        t_stop = self._max_sigs_length / self._max_sampling_rate
+        return t_stop
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+        stream_id = self.header["signal_streams"][stream_index]["id"]
+        mask = self.header["signal_channels"]["stream_id"] == stream_id
+        signal_channels = self.header["signal_channels"][mask]
+        channel_names = signal_channels["name"]
+        chan_name0 = channel_names[0]
+        if self.one_file_per_signal:
+            size = self._raw_data[stream_index][chan_name0].size
+        else:
+            size = self._raw_data[stream_index][0][chan_name0].size
+        return size
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        return 0.0
+
+    def _get_analogsignal_chunk(
+        self, block_index, seg_index, i_start, i_stop, stream_index, channel_indexes
+    ):
+
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self._get_signal_size(block_index, seg_index, stream_index)
+
+        stream_id = self.header["signal_streams"][stream_index]["id"]
+        mask = self.header["signal_channels"]["stream_id"] == stream_id
+        signal_channels = self.header["signal_channels"][mask]
+        if channel_indexes is None:
+            channel_indexes = slice(None)
+        channel_names = signal_channels["name"][channel_indexes]
+
+        if self.one_file_per_signal:
+            shape = self._raw_data[stream_index][channel_names[0]].shape
+        else:
+            shape = self._raw_data[stream_index][0][channel_names[0]].shape
+
+        # some channel (temperature) have 1D field so shape 1D
+        # because 1 sample per block
+        if len(shape) == 2:
+            # this is the general case with 2D
+            block_size = shape[1]
+            block_start = i_start // block_size
+            block_stop = i_stop // block_size + 1
+
+            sl0 = i_start % block_size
+            sl1 = sl0 + (i_stop - i_start)
+
+        sigs_chunk = np.zeros((i_stop - i_start, len(channel_names)), dtype="uint16")
+        for i, chan_name in enumerate(channel_names):
```

Comment:
I would avoid this `i`, you should use `channel_index`. There is already an `i` in the `i_start` so I think is abit confusing.